pkg/didcomm/protocol/exchange/service/service.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-openapi/runtime/middleware/denco"
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/exchange"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+	"github.com/hyperledger/aries-framework-go/pkg/internal/common/support"
+)
+
+var logger = log.New("aries-framework/did-exchange")
+
+const (
+	createInviationAPIPath = "/create-invitation"
+)
+
+// provider contains dependencies for the Exchange protocol and is typically created by using aries.Context()
+type provider interface {
```

Comment:
Suggest this interface be substituted by a `func() transport.OutboundTransport` instead (see #143)

Reply:
We are going to add more like inbound, state etc
Followed pattrern of `exchange.provider` interface

Reply:
Yes - `exchange.provider` is a single-method interface. I can't imagine a simple provider having more than one method, so I suggest doing away with the interface entirely and just declare dependency on the anonymous function

Reply:
Are we following up on this one?

Reply:
exchange.New(provider) needs `exchange.provider` interface to initialize.
So I redefined that interface in this package to avoid import cycle. And also `exchange.provider`is not exported.

Whoever changes `exchange.provider` interface will be forced to update here as it will cause compile errors.

Reply:
@troyronda well, it's just a suggestion.

Reply:
I'll merge to unblock people, but we should check that this conversation got resolved.